pkg/framework/retry.go

```diff
@@ -0,0 +1,35 @@
+package framework
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/cenkalti/backoff"
+)
+
+const maxElapsedTime = 2 * time.Minute
+
+// NewCustomExponentialBackoff returns pointer to a backoff.ExponentialBackOff,
+// initialized with custom values. At the moment, we only override the
+// maxElapsedTime.
+func NewCustomExponentialBackoff() *backoff.ExponentialBackOff {
+	b := &backoff.ExponentialBackOff{
+		InitialInterval:     backoff.DefaultInitialInterval,
+		RandomizationFactor: backoff.DefaultRandomizationFactor,
+		Multiplier:          backoff.DefaultMultiplier,
+		MaxInterval:         backoff.DefaultMaxInterval,
+		MaxElapsedTime:      maxElapsedTime,
+		Clock:               backoff.SystemClock,
+	}
+
+	b.Reset()
+
+	return b
+}
+
+func NewNotify(operationName string) func(error, time.Duration) {
+	return func(err error, delay time.Duration) {
+		log.Printf(fmt.Sprintf("%s failed, retrying with delay %.0fm%.0fs: '%#v'", operationName, delay.Minutes(), delay.Seconds(), err))
+	}
+}
```

Comment:
It doesn't need to be exported. I'm big on making everything private by default :)

Reply:
:100: